plugins/imfile/imfile.c

```diff
@@ -2667,46 +2796,268 @@ fen_processEvent(port_event_t* pportEvent, lstn_t *pLstn, int revents)
 		* here. Process received events.
 		*/
 		if (revents) {
-			/* File has been modified, trigger a pollFile */
 			if (revents & FILE_MODIFIED) {
+				/* File has been modified, trigger a pollFile */
 				pollFile(pLstn, NULL);
+			} else if (revents & FILE_RENAME_FROM) {
+				/* File has been renamed which means it was deleted. remove the file*/
+				fen_removeFile(pLstn);
+				FINALIZE;
 			}
 
 			/* If exception, no need to re-register. */
 			if (revents & FILE_EXCEPTION) {
+				fen_removeFile(pLstn);
```

Comment:
is this still an error case (ABORT in next stmt)?

Reply:
FILE_EXCEPTION is generated by the system when something really bad happened.  Unfortunetally there is no documentation on when FILE_EXCEPTION is generated. Just that it needs to be checked in the few code samples that can be found.

Reply:
I would add this as a comment. Saves questions ;-)

Reply:
added comment